backend/src/main/java/net/vidux/camhub/discovery/CameraScanViduxHelper.java

```diff
@@ -0,0 +1,34 @@
+package net.vidux.camhub.discovery;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+class CameraScanViduxHelper implements CameraScan {
+
+  @Autowired private ViduxHelperWrapper viduxHelperWrapper;
+
+  @Autowired private RawCameraDataFactory rawCameraDataFactory;
+
+  @Override
+  public CompletableFuture<Set<RawCameraData>> scanCams() {
+    List<String> rawCameraLineList;
+    try {
+      rawCameraLineList = viduxHelperWrapper.findHikvisionIpCameras();
+    } catch (IOException | TimeoutException e) {
+      return CompletableFuture.failedFuture(e);
+    }
+
+    return CompletableFuture.completedFuture(
+        rawCameraLineList.stream()
+            .map(rawCameraDataFactory::createRawCameraData)
+            .collect(Collectors.toCollection(HashSet::new)));
+  }
```

Comment:
Nekem itt az nem tetszik igazából, hogy ha minden rawCameraLine-ra meghívjuk a createRawCameraData-t, akkor az visszaadhat null-t és akkor egy olyan hashSet-ünk lesz, amiben lesznek RawCameraData példányok, plusz 1 db null érték (mivel a HashSet-be akárhányszor raksz null-t, 1 db null marad benne).

Valahogy úgy kellene elkerülni a null-okat, hogy eleve ne is lehessen olyan helyzet, hogy null-t állít elő a RawCameraDataFactory.

A legjobb szerintem lehet az lenne, ha a findHikvisionIpCameras metódus használna egy reguláris kifejezést, hogy kiszűrje az invalid sorokat és utána a RawCameraDataFactory-val létrehozná a sorok alapján a RawCameraData típusú objektumok listáját, ami egyébként inkább Set kéne, hogy legyen szerintem.